zkevm-circuits/src/pi_circuit.rs

```diff
@@ -0,0 +1,978 @@
+            _marker: PhantomData,
+        }
+    }
+
+    /// Return the number of rows in the circuit
+    #[inline]
+    fn circuit_len() -> usize {
+        BLOCK_LEN + EXTRA_LEN + 3 * (TX_LEN * MAX_TXS + MAX_CALLDATA)
+    }
+
+    /// Assigns a tx_table row and stores the values in a vec for the
+    /// raw_public_inputs column
+    #[allow(clippy::too_many_arguments)]
+    fn assign_tx_row(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        tx_id: usize,
+        tag: TxFieldTag,
+        index: usize,
+        tx_value: F,
+        raw_pi_vals: &mut Vec<F>,
+    ) -> Result<(), Error> {
+        let tx_id = F::from(tx_id as u64);
+        let tag = F::from(tag as u64);
+        let index = F::from(index as u64);
+        let tx_value = tx_value;
+
+        self.q_tx_table.enable(region, offset)?;
+
+        // Assign vals to Tx_table
+        region.assign_advice(|| "tx_id", self.tx_id, offset, || Ok(tx_id))?;
+        region.assign_fixed(|| "tag", self.tag, offset, || Ok(tag))?;
+        region.assign_advice(|| "index", self.index, offset, || Ok(index))?;
+        region.assign_advice(|| "tx_value", self.tx_value, offset, || Ok(tx_value))?;
+
+        // Assign vals to raw_public_inputs column
+        let tx_table_len = TX_LEN * MAX_TXS + MAX_CALLDATA;
+
+        let id_offset = BLOCK_LEN + EXTRA_LEN;
+        let index_offset = id_offset + tx_table_len;
+        let value_offset = index_offset + tx_table_len;
+
+        region.assign_advice(
+            || "raw_pi.tx_id",
+            self.raw_public_inputs,
+            offset + id_offset,
+            || Ok(tx_id),
+        )?;
+
+        region.assign_advice(
+            || "raw_pi.tx_index",
+            self.raw_public_inputs,
+            offset + index_offset,
+            || Ok(index),
+        )?;
+
+        region.assign_advice(
+            || "raw_pi.tx_value",
+            self.raw_public_inputs,
+            offset + value_offset,
+            || Ok(tx_value),
+        )?;
+
+        // Add copy to vec
+        raw_pi_vals[offset + id_offset] = tx_id;
+        raw_pi_vals[offset + index_offset] = index;
+        raw_pi_vals[offset + value_offset] = tx_value;
+
+        Ok(())
+    }
+
+    /// Assigns the values for block table in the block_table column
+    /// and in the raw_public_inputs column. A copy is also stored in
+    /// a vector for computing RLC(raw_public_inputs)
+    fn assign_block_table(
+        &self,
+        region: &mut Region<'_, F>,
+        block_values: BlockValues,
+        randomness: F,
+        raw_pi_vals: &mut Vec<F>,
+    ) -> Result<AssignedCell<F, F>, Error> {
+        let mut offset = 0;
+        for i in 0..BLOCK_LEN {
+            self.q_block_table.enable(region, offset + i)?;
+        }
+
+        // coinbase
+        let mut coinbase_bytes = [0u8; 32];
+        coinbase_bytes[12..].clone_from_slice(block_values.coinbase.as_bytes());
+        let coinbase = rlc(coinbase_bytes, randomness);
```

Comment:
Since coinbase is an address, it already fits in a field and should not be rlc encoded.

The same applies to all the types that should be u64.

NOTE: you may have noticed that in the spec, there's no value which is rlc encoded (which means that some values wrap around due to the field being less than 256 bits).  The reason for that is that when I was specifying the PublicInputs Circuit, it was not clear if we were going to do https://github.com/privacy-scaling-explorations/zkevm-specs/issues/216 (which removes the RLC for words) before the first milestone or afterwards.  Which means that the spec is currently incomplete with missing details :P

Reply:
You're right! In fact there is a `to_scalar()` function for `Address` so these lines are replaced for that function  call